app/forms.py

```diff
@@ -130,6 +130,45 @@ class EstateAgentRequestForm(FlaskForm):
 
     submit = SubmitField("Submit Request")
 
+class HallsRequestForm(FlaskForm):
+    lat = None
+    lng = None
+    errors = []
+
+    name = StringField("Halls Name", validators=[DataRequired(), Length(min=2, max=58)])
+    address_line_1 = StringField("Address Line 1", validators=[Length(min=2, max=94)])
+    address_line_2 = StringField("Address Line 2", validators=[Length(max=94)])
+    city = StringField("City", validators=[Length(min=2, max=58)])
+    postcode = StringField("Postcode", validators=[Length(min=5, max=8), Regexp("([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})", message="Invalid postcode")])
+    website = StringField("Halls Website", validators=[DataRequired()])
+    rent = PriceField("Rent £", validators=[DataRequired(), NumberRange(min=0, max=10000)])
+    comment = TextAreaField("Comment")
+
+    university = RadioField("Which university do you belong to?", choices=[('cardiff_uni','Cardiff University'),('cardiff_met','Cardiff Metropolitan University'),('usw','University of South Wales')], validators=[DataRequired()])
+    first_name = StringField("First Name", validators=[DataRequired(), Length(min=2, max=58)])
+    last_name = StringField("Last Name", validators=[DataRequired(), Length(min=2, max=58)])
+    user_email = StringField("Email", validators=[DataRequired()])
```

Comment:
Maybe add regex for email and website as well as max limits for comments

Reply:
I'll add it as part of a larger issue as none of the forms at the moment have regex for emails yet

Reply:
ah nice one 👍